assets/js/admin/meta-boxes-product-variation.js

```diff
@@ -469,7 +470,12 @@ jQuery( function( $ ) {
 					return;
 				}
 
-				data[ elem.name ] = elem.value;
+				// Do not send values for empty multi-selects.
+				if ( $input.is( 'select[multiple]' ) && null === value ) {
```

Comment:
Looks great!

How would you feel about unwrapping the array instead of assigning it to the data object? None of the other input types feature nesting and so having it for this input might be a little strange. The only trouble with that would be that you need to set the array indexing manually; replacing the `[]` in the element's name with a correct index.

Reply:
I guess this extends to duplicates too. With a normal form, `key[]` can be used to populate an array with multiple inputs. With this however we would set each one to the later's key. Perhaps instead of unwrapping it like I suggested we just keep arrays and handle duplicates with them?